src/test/java/com/wojciechkocik/usage/service/CourseUsageServiceTest.java

```diff
@@ -0,0 +1,217 @@
+        CourseUsage courseUsageWithTimeCrossedMidnight = new CourseUsage();
+        ZonedDateTime startedOneMinuteBeforeMidnight = ZonedDateTime.parse("2017-03-23T23:59:00.000+01:00[Europe/Warsaw]");
+        String firstSimpleDay = "2017-03-23";
+        String secondSimpleDay = "2017-03-24";
+        courseUsageWithTimeCrossedMidnight.setStarted(startedOneMinuteBeforeMidnight);
+        courseUsageWithTimeCrossedMidnight.setCourseId(courseId);
+        courseUsageWithTimeCrossedMidnight.setUserId(fairy.textProducer().randomString(10));
+        courseUsageWithTimeCrossedMidnight.setTimeSpent(Duration.ofMinutes(timeSpentMinutes).getSeconds()); //one minute in the next day
+
+        courseUsageRepository.save(courseUsageWithTimeCrossedMidnight);
+
+        int responseSizeExpected = 2;
+
+        long firstDayMinutesExpected = 1;
+        long secondDayMinutesExpected = timeSpentMinutes - firstDayMinutesExpected;
+
+        //Act
+        List<DailyUsageResponse> dailyUsagesForCourse = courseUsageService.findDailyUsageForCourse(courseId);
+        int responseSizeActual = dailyUsagesForCourse.size();
+        long firstDayMinutesActual = dailyUsagesForCourse.stream().filter(f -> f.getDateTime().equals(firstSimpleDay))
+                .collect(Collectors.toList()).get(0).getTime();
+        long secondDayMinutesActual = dailyUsagesForCourse.stream().filter(f -> f.getDateTime().equals(secondSimpleDay))
+                .collect(Collectors.toList()).get(0).getTime();
+
+        //convert to minutes
+        firstDayMinutesActual = Duration.ofSeconds(firstDayMinutesActual).toMinutes();
+        secondDayMinutesActual = Duration.ofSeconds(secondDayMinutesActual).toMinutes();
+
+        //Assert
+        Assert.assertEquals(responseSizeExpected, responseSizeActual);
+        Assert.assertEquals(firstDayMinutesExpected, firstDayMinutesActual);
+        Assert.assertEquals(secondDayMinutesExpected, secondDayMinutesActual);
+    }
+
+    @Test
+    public void findDailyUsageForCourse_groupDate_hasProperSpentTime() {
+        //Arrange
+        ZonedDateTime started = ZonedDateTime.parse("2017-03-24T11:56:26.595+01:00[Europe/Belgrade]");
+        int entitiesWithSameDateForGroup = 5;
+        String userId = fairy.textProducer().randomString(10);
+        String courseId = fairy.textProducer().randomString(10);
+
+        for (int i = 0; i < entitiesWithSameDateForGroup; i++) {
+            CourseUsage courseUsage = new CourseUsage(
+                    started,
+                    new Random().nextInt(5000),
+                    userId,
+                    courseId
+            );
+            courseUsageRepository.save(courseUsage);
+        }
+
+        int sizeExpected = 1;
+
+        //Act
+        List<DailyUsageResponse> dailyUsageForCourse = courseUsageService.findDailyUsageForCourse(courseId);
+        int sizeActual = dailyUsageForCourse.size();
+
+        //Assert
+        Assert.assertEquals(sizeExpected, sizeActual);
```

Comment:
The test say that it tests that the groupDate hasProperSpentTime, but you tests whether the service returned correct number of DailyUsageResponses. Maybe it works, but is very very implicit!